R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,380 @@
+      FAO2050_Diet
+    L100.FAO_ag_Food_t <- get_data(all_data, "L100.FAO_ag_Food_t")
+    L100.FAO_an_Food_t <- get_data(all_data, "L100.FAO_an_Food_t")
+    L101.ag_Food_Pcal_R_C_Y <- get_data(all_data, "L101.ag_Food_Pcal_R_C_Y")
+    L105.an_Food_Pcal_R_C_Y <- get_data(all_data, "temp-data-inject/L105.an_Food_Pcal_R_C_Y") %>%
+      # TEMPORARY - for temp-data-inject data
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity) %>% mutate(year = as.integer(substr(year, 2, 5)))
+    L101.Pop_thous_R_Yh <- get_data(all_data, "L101.Pop_thous_R_Yh")
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+
+    # Build historical time series of per-capita caloric demands
+
+    # Historical time series of ag and animal product consumption
+    # Original lines 41-49
+    L101.ag_Food_Pcal_R_C_Y %>%
+      filter(year %in% AGLU_HISTORICAL_YEARS) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "crops") ->
+      L134.ag_Food_Pcal_R_Y
+
+    L105.an_Food_Pcal_R_C_Y %>%
+      filter(year %in% AGLU_HISTORICAL_YEARS) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.ag_Food_Pcal_R_Y) ->
+      L134.Food_Pcal_R_Dmnd_Y
+
+    # Divide by population to calculate the historical per-capita food demands, in kcal per person per day
+    # Original lines 51-56
+    L134.Food_Pcal_R_Dmnd_Y %>%
+      left_join_error_no_match(L101.Pop_thous_R_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * CONV_DAYS_YEAR * (1 / CONV_MCAL_PCAL ) / value.y) %>%
+      select(-value.x, -value.y) ->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # Extrapolate this to the future periods based on the FAO projections
+    # Original lines 58-69
+    L100.FAO_ag_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "crops") ->
+      L134.FAO_ag_Food_t
+
+    L100.FAO_an_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.FAO_ag_Food_t) %>%
+      left_join_keep_first_only(select(AGLU_ctry, FAO2050_reg, iso), by = "iso") ->
+      L134.Food_t_ctry_Dmnd_Y
+
+    # Calculate the future demand ratios by FAO2050 region for each demand type
+    # Drop unnecessary composite regions from FAO diet table
+    # Original lines 71-85
+    FAO2050_Diet %>%
+      filter(FAO2050_reg %in% L134.Food_t_ctry_Dmnd_Y$FAO2050_reg) %>%
+      # Calculate FAO2050 diet for specified diet years
+      # Use linear interpolation to convert FAO2050 model time periods to GCAM "diet years"
+      group_by(FAO2050_reg, FAO2050_item) %>%
+      # Extend years in dataset to include DIET_YEARS
+      complete(year = unique(c(.$year, aglu.DIET_YEARS))) %>%
```

Comment:
is this different from `tidyr::complete`?  Should it be? It doesn't look like `complete` was imported from `tidyr` with `gather` and `spread` in the header.